bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/TccFieldConfig.java

```diff
@@ -53,21 +53,16 @@ public boolean build() throws WdbaGlueException {
         putParameter(TccNames.GUIDE_GROUP, getPrimaryGuideGroupName());
 
         // Add a rotator parameter and then create the config
-        // Note: If the config name contains "Fixed" in some way, it's assumed to be a Fixed rotator config
-        RotatorConfig rc = new RotatorConfig(_oe);
+        final RotatorConfig rc = new RotatorConfig(_oe);
         if (rc.build()) {
-            String rotConfigName = rc.getConfigName();
-            if (rotConfigName.equals(TccNames.ALTAIR_FIXED)) {
-                putParameter(TccNames.ROTATOR, rotConfigName);
-            } else
             // If the instrument adds something, create a rotator config and add it in
-            if (rotConfigName.contains(TccNames.FIXED)) {
-                putParameter(TccNames.ROTATOR, rotConfigName);
-                add(rc);
-            } else {
-                putParameter(TccNames.POSANGLE, rotConfigName);
-                add(rc);
-            }
+            ImOption.apply(rc.attributeValue(TYPE)).foreach(t -> {
+                final String name = rc.getConfigName();
+                putParameter(t, name);
+                if (!TccNames.ALTAIR_FIXED.equals(name)) {
```

Comment:
I don't know the reasoning behind skipping the rotator config param set addition for `AltairFixed` but I won't question it.  I assume it must refer to a file that gets loaded in the TCC or something.